### PR TITLE
Ft/nested routes

### PR DESCRIPTION
**Pull request by serapieTuyishime**
Implement :

1. All vans visualization
2. navigation through the whole page
3. implementing nested and relative routes
4. adding basic styling

Missing: 

1. Applying placeholder components
2. Adding error handling

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { Vans as HostVans } from "./pages/Host/Vans";
+import { VanDetails as HostVandetails } from "./pages/Host/VanDetails";
+import { SingleVanDetails as Details } from "./components/vans/SingleVanDetails";
+import { Pricing } from "./components/vans/Pricing";
+import Photos from "./components/vans/Photos";
+
+function App() {
+    return (
+        <BrowserRouter>
+            <Navigation />
```
Comment (izzett222) [inline src/App.jsx:24]: Could you move the navigation inside a layout route?

File: src/components/Cards/Van.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
```
Comment (izzett222) [inline src/components/Cards/Van.jsx]: You don't need to import React.

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            <Routes>
+                <Route path="/" element={<HomePage />} />
+                <Route path="/about" element={<About />} />
+                <Route path="/vans" element={<VansList />} />
+                <Route path="/vans/:id" element={<VanDetails />} />
+                <Route path="/host" element={<HostLayout />}>
+                    <Route index element={<Dashboard />} />
+                    <Route path="dashboard" element={<Dashboard />} />
```
Comment (izzett222) [inline src/App.jsx]: The index route is the dashboard, so there is no need to add another route for /dashboard.

File: src/components/ui/HostNavigation.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const HostNavigation = () => {
+    const activeStyles = {
+        fontWeight: "bold",
+        textDecoration: "underline",
+        color: "#161616",
+    };
+    return (
+        <div className="flex justify-start gap-4 item-center">
+            <NavLink
+                to="/host/dashboard"
```
Comment (izzett222) [inline src/components/ui/HostNavigation.jsx]: You can use a relative path to link to the dashboard, as it is the index route.
```suggestion
                to="."
```

File: src/components/ui/HostNavigation.jsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                className={`text-2xl`}
+                style={({ isActive }) => (isActive ? activeStyles : null)}
+            >
+                Dashboard
+            </NavLink>
+            <NavLink
+                to="/host/income"
```
Comment (izzett222) [inline src/components/ui/HostNavigation.jsx]: You can also use a relative path for these nested routes.

File: src/components/ui/HostNavigation.jsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                className="text-2xl"
+                style={({ isActive }) => (isActive ? activeStyles : null)}
+            >
+                Income
+            </NavLink>
+            <NavLink
+                to="/host/reviews"
```
Comment (izzett222) [inline src/components/ui/HostNavigation.jsx]: You can also use a relative path for these nested routes.

File: src/components/ui/HostNavigation.jsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                className="text-2xl"
+                style={({ isActive }) => (isActive ? activeStyles : null)}
+            >
+                Reviews
+            </NavLink>
+            <NavLink
+                to="/host/vans"
```
Comment (izzett222) [inline src/components/ui/HostNavigation.jsx]: You can also use a relative path for these nested routes.

File: src/components/ui/TitleElement.jsx [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
Comment (izzett222) [inline src/components/ui/TitleElement.jsx]: You don't need to import React.

File: src/pages/Host/Dashboard.jsx [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
Comment (izzett222) [inline src/pages/Host/Dashboard.jsx]: You don't need to import React.

File: src/pages/Host/Income.jsx [hunk 10]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
Comment (izzett222) [inline src/pages/Host/Income.jsx]: You don't need to import React.

File: src/pages/Host/Reviews.jsx [hunk 11]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
```
Comment (izzett222) [inline src/pages/Host/Reviews.jsx]: You don't need to import React.

File: src/pages/Host/VanDetails.jsx [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+                            <div className="flex items-center gap-3">
+                                <Link className="" to={`details`}>
+                                    Details
+                                </Link>
+                                <Link className="" to={`pricing`}>
+                                    Pricing
+                                </Link>
+                                <Link className="" to={`photos`}>
+                                    Photos
+                                </Link>
```
Comment (izzett222) [inline src/pages/Host/VanDetails.jsx:58]: Could you remove the empty className?

File: src/App.jsx (hunk 1 above)
Comment (serapieTuyishime) [reply src/App.jsx:24]: It seems hard for me. I will do it in the finish ups.

**Review by izzett222 — APPROVED** [review]

**Pull request merged**